packages/vscode-extension/src/commandHandlers.ts

```diff
@@ -192,8 +192,20 @@ export async function runCypher(): Promise<void> {
       return;
     }
 
-    const documentText = editor.document.getText();
+    const selections = editor.selections.filter(
+      (selection) => !selection.isEmpty && editor.document.getText(selection),
+    );
+
     const documentUri = editor.document.uri;
-    await cypherRunner.run(activeConnection, documentUri, documentText);
+
+    if (selections.length === 0) {
+      const text = editor.document.getText();
+      await cypherRunner.run(activeConnection, documentUri, Array.of(text));
```

Comment:
It works but I've never seen anyone do `Array.of(stringValue)` instead of `[stringValue]`